src/overlays/actors/ovl_En_Invisible_Ruppe/z_en_invisible_ruppe.c

```diff
@@ -30,25 +30,93 @@ const ActorInit En_Invisible_Ruppe_InitVars = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_NONE,
+        OC1_ON | OC1_NO_PUSH | OC1_TYPE_PLAYER,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+
+    {
+        ELEMTYPE_UNK4,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000000, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
+
     { 10, 30, 0, { 0, 0, 0 } },
 };
 
-#endif
+void func_80C258A0(EnInvisibleRuppe* this, GlobalContext* globalCtx) {
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 32.0f, 30.0f, 0.0f, 4);
+}
+
+void func_80C2590C(EnInvisibleRuppe* this, GlobalContext* globalCtx) {
+    if ((this->collider.base.ocFlags1 & 2) != 0) {
+        switch (this->actor.params & 3) {
+            case 0:
+                play_sound(NA_SE_SY_GET_RUPY);
+                Item_DropCollectible(globalCtx, &this->actor.world.pos, 0x8000);
+                break;
+
+            case 1:
+                play_sound(NA_SE_SY_GET_RUPY);
+                Item_DropCollectible(globalCtx, &this->actor.world.pos, 0x8001);
+                break;
+
+            case 2:
+                play_sound(NA_SE_SY_GET_RUPY);
+                Item_DropCollectible(globalCtx, &this->actor.world.pos, 0x8002);
+                break;
+        }
+
+        if (this->unk_190 >= 0) {
+            Actor_SetSwitchFlag(globalCtx, this->unk_190);
+        }
+
+        this->actionFunc = func_80C259E8;
+    }
+}
+
+void func_80C259E8(EnInvisibleRuppe* this, GlobalContext* globalCtx) {
+    Actor_MarkForDeath(&this->actor);
+}
+
+void EnInvisibleRuppe_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnInvisibleRuppe* this = THIS;
+    s32 pad;
 
-extern ColliderCylinderInit D_80C25B50;
+    this->unk_190 = (this->actor.params & 0x1FC) >> 2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Invisible_Ruppe/func_80C258A0.s")
+    if (this->unk_190 == 0x7F) {
+        this->unk_190 = -1;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Invisible_Ruppe/func_80C2590C.s")
+    if (this->unk_190 >= 0 && Flags_GetSwitch(globalCtx, this->unk_190) != 0) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Invisible_Ruppe/func_80C259E8.s")
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Invisible_Ruppe/EnInvisibleRuppe_Init.s")
+    this->actionFunc = &func_80C2590C;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Invisible_Ruppe/EnInvisibleRuppe_Destroy.s")
+void EnInvisibleRuppe_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnInvisibleRuppe* this = THIS;
+    Collider_DestroyCylinder(globalCtx, &this->collider);
```

Comment:
```suggestion
    EnInvisibleRuppe* this = THIS;

    Collider_DestroyCylinder(globalCtx, &this->collider);
```
We always add spaces between declarations and the rest of the code.